satpy/tests/reader_tests/test_grib.py

```diff
@@ -205,85 +283,29 @@ def test_file_pattern(self):
-        ])
-        r.create_filehandlers(loadables)
-        datasets = r.load([
+        fake_pygrib = self._get_fake_pygrib(proj_params, lon_corners, lat_corners)
+        dataids = [
             DataQuery(name='t', level=100, modifiers=tuple()),
             DataQuery(name='t', level=200, modifiers=tuple()),
-            DataQuery(name='t', level=300, modifiers=tuple())])
-        self.assertEqual(len(datasets), 3)
-        for v in datasets.values():
-            self.assertEqual(v.attrs['units'], 'K')
-            self.assertIsInstance(v, xr.DataArray)
-            self._round_trip_projection_lonlat_check(v.attrs['area'])
-
-    @mock.patch('satpy.readers.grib.pygrib')
-    def test_load_all_lcc(self, pg):
-        """Test loading all test datasets with lcc projections."""
-        lats = np.array([
-            [12.19, 0, 0, 0, 14.34208538],
-            [0, 0, 0, 0, 0],
-            [0, 0, 0, 0, 0],
-            [0, 0, 0, 0, 0],
-            [54.56534318, 0, 0, 0, 57.32843565]])
-        lons = np.array([
-            [-133.459, 0, 0, 0, -65.12555139],
-            [0, 0, 0, 0, 0],
-            [0, 0, 0, 0, 0],
-            [0, 0, 0, 0, 0],
-            [-152.8786225, 0, 0, 0, -49.41598659]])
-        pg.open.return_value = FakeGRIB(
-            proj_params={
-                'a': 6371229, 'b': 6371229, 'proj': 'lcc',
-                'lon_0': 265.0, 'lat_0': 25.0,
-                'lat_1': 25.0, 'lat_2': 25.0},
-            latlons=(lats, lons))
-        from satpy.readers import load_reader
-        r = load_reader(self.reader_configs)
-        loadables = r.select_files_from_pathnames([
-            'gfs.t18z.sfluxgrbf106.grib2',
-        ])
-        r.create_filehandlers(loadables)
-        datasets = r.load([
-            DataQuery(name='t', level=100, modifiers=tuple()),
-            DataQuery(name='t', level=200, modifiers=tuple()),
-            DataQuery(name='t', level=300, modifiers=tuple())])
-        self.assertEqual(len(datasets), 3)
+            DataQuery(name='t', level=300, modifiers=tuple())
+        ]
+        datasets = self._get_test_datasets(dataids, fake_pygrib)
+
+        assert len(datasets) == 3
         for v in datasets.values():
-            self.assertEqual(v.attrs['units'], 'K')
-            self.assertIsInstance(v, xr.DataArray)
-            self._round_trip_projection_lonlat_check(v.attrs['area'])
+            assert v.attrs['units'] == 'K'
+            assert isinstance(v, xr.DataArray)
+
+    @pytest.mark.parametrize(TEST_ARGS, TEST_PARAMS)
+    def test_area_def_crs(self, proj_params, lon_corners, lat_corners):
+        """Check that the projection is accurate."""
+        fake_pygrib = self._get_fake_pygrib(proj_params, lon_corners, lat_corners)
+        dataids = [DataQuery(name='t', level=100, modifiers=tuple())]
+        datasets = self._get_test_datasets(dataids, fake_pygrib)
+        area = datasets['t'].attrs['area']
+        _round_trip_projection_lonlat_check(area)
```

Comment:
```suggestion
        area = datasets['t'].attrs['area']
        if not hasattr(area, 'crs'):
            pytest.skip("Can't test with pyproj < 2.0")
        _round_trip_projection_lonlat_check(area)
```

Reply:
Good idea. I'd like to move it inside the function though. Hhhmm, although I suppose this is specific to this test so maybe you do have the best place for it.